lib/MongoDB/Role/_SessionModifier.pm

```diff
@@ -0,0 +1,72 @@
+#
+#  Copyright 2014 MongoDB, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+use strict;
+use warnings;
+package MongoDB::Role::_SessionModifier;
+
+# MongoDB role to manage clusterTime commands
+
+use Moo::Role;
+use MongoDB::Error;
+use MongoDB::_Types -types, 'to_IxHash';
+
+use namespace::clean;
+
+requires qw/client/;
```

Comment:
Ditto: I think we should access 'client' from the session, if there is one.  See comment below about validation of client.